nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/event/pipe/queue/QueueBaseEventBundlePipe.java

```diff
@@ -64,9 +61,8 @@ public void initPipe(String name, Map<String, String> params) {
                 log.error("Unable to read batchSize parameter", e);
             }
         }
-
         queue = new ConcurrentLinkedQueue<>();
-        consumerTPE = new ThreadPoolExecutor(1, 1, 60, TimeUnit.MINUTES, new LinkedBlockingQueue<Runnable>());
+        consumerTPE = new ThreadPoolExecutor(1, 1, 60, TimeUnit.MINUTES, new LinkedBlockingQueue<>());
```

Comment:
The TP management is a bit too fragile;
- the worker thread should not manage TPE
- there are Thread.sleep
- the unbounded queue can turn into OOM if the consumer is slow 
- introducing another in memory queue is a source of weakness we may loose messages here out of a transaction